File: THCS/2022_2023/QuangNinh_2223/Bai1/disp.py
import math

# Đọc dữ liệu từ file
with open('disp.inp', 'r') as f:
    n = int(f.read().strip())

# Biến lưu cặp (a, b) thỏa mãn với hiệu b - a nhỏ nhất
best_a, best_b = 1, n

# Duyệt từ 1 đến sqrt(n) để tìm các ước số của n
for a in range(1, int(math.sqrt(n)) + 1):
    if n % a == 0:
        b = n // a
        # Kiểm tra điều kiện a <= b và cập nhật nếu b - a nhỏ hơn
        if a <= b and b - a < best_b - best_a:
            best_a, best_b = a, b

# Ghi kết quả ra file
with open('disp.out', 'w') as f:
    f.write(f'{best_a} {best_b}\n')

File: THCS/2022_2023/QuangNinh_2223/Bai1/disp.cpp
#include <iostream>
#include <fstream>
#include <cmath>

using namespace std;

int main() {
    // Mở file input để đọc dữ liệu
    ifstream infile("disp.inp");
    ofstream outfile("disp.out");

    long long n;
    infile >> n; // Đọc giá trị n

    long long best_a = 1, best_b = n; // Biến lưu cặp (a, b) tốt nhất ban đầu

    // Duyệt các giá trị a từ 1 đến sqrt(n)
    for (long long a = 1; a * a <= n; ++a) {
        if (n % a == 0) { // Nếu a là ước của n
            long long b = n / a;
            if (a <= b && b - a < best_b - best_a) {
                // Cập nhật nếu tìm được cặp (
